src/resolvelib/providers.py

```diff
@@ -48,17 +48,19 @@ def get_preference(self, resolution, candidates, information):
         """
         raise NotImplementedError
 
-    def find_matches(self, requirement):
-        """Find all possible candidates that satisfy a requirement.
+    def iter_matches(self, requirements):
+        """Find all possible candidates that satisfy all given requirements.
```

Comment:
```suggestion
        """Find all possible candidates that satisfy the given requirements.
```
Reads better without the repeated "all" in the sentence.